packages/sanity/src/core/components/InsufficientPermissionsMessage.tsx

```diff
@@ -2,47 +2,49 @@ import {AccessDeniedIcon} from '@sanity/icons'
 import {CurrentUser} from '@sanity/types'
 import {Text, Inline, Box} from '@sanity/ui'
 import React from 'react'
+// note: these are both available from the `../i18n` export but importing through
+// that export fails the build. may be due to a circular reference.
+import {useTranslation} from '../i18n/hooks/useTranslation'
+import {Translate} from '../i18n/Translate'
 
 /** @internal */
 export interface InsufficientPermissionsMessageProps {
-  title?: string
   operationLabel?: string
   currentUser?: CurrentUser | null
 }
 
 /** @internal */
 export function InsufficientPermissionsMessage(props: InsufficientPermissionsMessageProps) {
+  const {t} = useTranslation()
   const {
     currentUser,
-    title = 'Insufficient permissions',
-    operationLabel = 'access this feature',
+    operationLabel = t('insufficient-permissions-message.default-operation-label'),
   } = props
 
   const roles = currentUser?.roles || []
-  const plural = roles.length !== 1
 
   return (
     <Box padding={2}>
       <Inline space={2}>
         <Text size={1}>
           <AccessDeniedIcon />
         </Text>
-        <Text weight="semibold">{title}</Text>
+        <Text weight="semibold">{t('insufficient-permissions-message.title')}</Text>
       </Inline>
       <Inline marginTop={4}>
         <Text size={1}>
-          {roles.length === 0 ? (
-            <>You have no role that grants you permission to {operationLabel}</>
-          ) : (
-            <>
-              Your role{plural && 's'}{' '}
-              {join(
-                roles.map((r) => <code key={r.name}>{r.title}</code>),
-                <>, </>,
-              )}{' '}
-              do{plural || 'es'} not have permissions to {operationLabel}
-            </>
-          )}
+          <Translate
+            i18nKey="insufficient-permissions-message.roles"
```

Comment:
To allow more control of the sentences, I would actually prefer if we broke this into separate translation keys instead of injecting `operationLabel` as a interpolation value. Yes, it will be a bit repetitive, but I think it is better to give translators full freedom of the sentence building.

Reply:
sounds good. i was about to go that route but the change was bigger but i'll give it another go!